packages/at_secondary_server/lib/src/verb/handler/pkam_verb_handler.dart

```diff
@@ -82,14 +74,99 @@ class PkamVerbHandler extends AbstractVerbHandler {
       response.errorMessage = 'pkam publickey not found';
       throw UnAuthenticatedException('pkam publickey not found');
     }
-    var isValidSignature = false;
+    var isValidSignature =
+        await _validateSignature(verbParams, sessionID, atSign, publicKey);
+    // authenticate if signature is valid
+    if (isValidSignature) {
+      atConnectionMetadata.isAuthenticated = true;
+      atConnectionMetadata.authType = pkamAuthType;
+      atConnectionMetadata.enrollmentId = enrollId;
+      response.data = 'success';
+    } else {
+      atConnectionMetadata.isAuthenticated = false;
+      response.data = 'failure';
+      logger.severe('pkam authentication failed');
+      throw UnAuthenticatedException('pkam authentication failed');
+    }
+  }
 
-    //retrieve stored secret using sessionid and atsign
-    var storedSecret = await keyStore.get('private:$sessionID$atSign');
-    storedSecret = storedSecret?.data;
+  /// Performs operations required for apkam authentication
+  /// Returns pkamPublicKey if the [enrollId] is [EnrollStatus.approved]
+  /// Otherwise returns [Response] with appropriate [Response.errorCode] and [Response.errorMessage]
+  /// Throws [KeyNotFoundException] if the key for provided [enrollId] does not exist
+  @visibleForTesting
+  Future<dynamic> handleApkamVerification(
```

Comment:
remove future dynamic and have an internal object.
Also remove response object from input param
e.g 
class ApkamVerificationResult {
Response response;
String apkamPublicKey;
}

Reply:
changes made as part of [7566e7e](https://github.com/atsign-foundation/at_server/pull/1520/commits/7566e7e79435c00f0b11a050b6cbc72ebbf54487)